test/login-throttle.js

```diff
@@ -0,0 +1,151 @@
+const t = require('../test-lib/test.js');
+const assert = require('assert');
+const Promise = require('bluebird');
+
+let apos;
+
+describe('Login', function() {
+
+  this.timeout(20000);
+
+  after(function(done) {
+    return t.destroy(apos, done);
+  });
+
+  // EXISTENCE
+
+  it('should initialize', function(done) {
+    apos = require('../index.js')({
+      root: module,
+      shortName: 'test',
+      modules: {
+        'apostrophe-express': {
+          secret: 'xxx',
+          port: 7901,
+          csrf: false
+        },
+        'apostrophe-users': {
+          groups: [
+            {
+              title: 'guest',
+              permissions: ['guest']
+            },
+            {
+              title: 'admin',
+              permissions: ['admin']
+            }
+          ],
+          disableInactiveAccounts: {
+            inactivityDuration: 0
+          }
+        },
+        'apostrophe-login': {
+          throttle: {
+            attempts: 3,
+            per: 0.25,
+            lockout: 0.25
+          }
+        }
+      },
+      afterInit: function(callback) {
+        assert(apos.modules['apostrophe-login']);
+        apos.argv._ = [];
+        assert(apos.users.safe.remove);
+        return apos.users.safe.remove({}, callback);
+        // return callback(null);
+      },
+      afterListen: function(err) {
+        assert(!err);
+        done();
+      }
+    });
+  });
+
+  it('should be able to insert test user', function(done) {
+    assert(apos.users.newInstance);
+    const user = apos.users.newInstance();
+    assert(user);
+
+    user.firstName = 'Harry';
```

Comment:
I mean 'crookshanks' should go with Hermione, but I digress.

Reply:
And I should replace this well-worn example with an author who hasn't proven herself a stubbornheaded transphobe. ):

Reply:
Also true. "Lilith Iyapo" and "nikanj" for a semi-deep cut (from a superb book).